### ISSUE TITLE
git.latest returning fatal error message

### ISSUE THREAD

**Issue by noelmcloughlin** [saltstack/salt#51157]
### Description of Issue/Bug

How can this error be disabled - its noise and git.latest succeeds.

```
[ERROR   ] Command '[u'git', u'rev-parse', u'--show-toplevel']' failed with return code: 128
[ERROR   ] stderr: fatal: not a git repository (or any of the parent directories): .git
[ERROR   ] retcode: 128
```

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

State is here:
https://github.com/saltstack-formulas/devstack-formula/blob/2f3f48240f29c7f97f6a0a2a7848c9bf006fb5ed/devstack/install.sls#L25

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

Run this state.
```
devstack.install
```

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2018.3.3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: unknown
       dateutil: 2.6.1
      docker-py: 3.7.0
          gitdb: 2.0.3
      gitpython: 2.1.8
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: 1.3.10
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.15rc1 (default, Nov 12 2018, 14:31:15)
   python-gnupg: 0.4.1
         PyYAML: 3.13
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: Ubuntu 18.04 bionic
         locale: ISO-8859-1
        machine: x86_64
        release: 4.15.0-39-generic
         system: Linux
        version: Ubuntu 18.04 bionic
```

**Comment by dwoz** [comment]
@noelmcloughlin I don't know of a way to silence this log. If the state is working, what is the driver for wanting to silence the log?

**Comment by noelmcloughlin** [comment]
Many reasons. Software quality; user experience; confusing information; uncertainty; trust; 
and to flip the coin on the question posed, if state is working why inform users on stdout a fatal Error occurred.   Did `git.latest` corrupt the download so that is why it was '**fatal**'?  How can I have 100% confidence the state worked - something went wrong presumably.

**Comment by dwoz** [comment]
Something did indeed go wrong, that is why I am on the fence about just changing these logs to the warn level or suppress them all together. A better solution would probably be for someone to dig in and figure out why the state is running that command and only run it when it's expected to pass, failing the state if it does not.

**Comment by noelmcloughlin** [comment]
I agree 100% @dwoz thanks for reviewing and the input. I'll change the title to indicate its a bug for tracking and  someone in the community might replicate.

**Comment by pavel-z1** [comment]
Up

This error occurred when git repository start clone first time. This mean when destination directory is empty Salt always trigger error.
This error confuse because operation as result executed successfully.

I propose to add additional checks to the state.latest and state.clone that will prevent execute command "git rev-parse --show-toplevel" when destination directory currently empty.

**Comment by sagetherage** [comment]
@dwoz looks like this isn't an open question any longer, but a bug, and do I read it correctly for the community to attempt to fix?

**Comment by pavel-z1** [comment]
The main problem for me with this bug, that I am receive report to the Sentry from Salt minion each time when git is cloned to the server first time.

Simple check for the empty destination folder before execute command '[u'git', u'rev-parse', u'--show-toplevel']' will prevent error messages.

**Comment by vrubiella** [comment]
+1 and same scenario with svn.latest
